### DISCUSSION TITLE
Could we have doOnAnything operator?

### DISCUSSION THREAD

**Discussion by Thaina** [ReactiveX/rxjs#6476] [Ideas / Feature request]
Suppose there was an ajax call to server that, when it was trigger, it should disable many UI in the page. Normally we could use `Do` with `onNext` and `onError` to enable the page again whenever the result was coming or throwing error. But if there was just the same long line of codes it became 2 duplicate blocks

So I would like to propose this operator to catch all to do something. It would have no input like a `Finally` but trigger just before next step

```js
Rx.Observable.range(0, 3)
  .do(
    function (x)   {
        // long line of code to enable many places
    },
    function (err) {
        // same long line of code to enable many places as above
    },
);

// instead

Rx.Observable.range(0, 3)
  .doOnAnything(
    function () {
        // same long line of code to enable many places as above
    },
);
```

Is it possible?

**Comment by martinsik** [comment]
You can very easily implement this yourself and this probably isn't a very common use-case so it's very unlikely this could become part of RxJS core.
